### PR TITLE
Add documentation update workflow

### PR DESCRIPTION
**Pull request by developomp**
Added GitHub workflow to automatically update the documentation.

### CODE REVIEW

File: .github/workflows/generate_docs.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      - name: Generate documentation
+        run: |
+          pdoc -o ./docs pytermgui
+
+      - name: Commit changes
+        run: |
+          git config --local user.email "41898282+github-actions[bot]@users.noreply.github.com"
+          git config --local user.name "github-actions[bot]"
+          git add ./docs/
+          git commit -m "auto documentation update"
```
Comment (bczsalba) [inline .github/workflows/generate_docs.yml:37]: The project's commit messages should be capitalized and uses an imperative style. Could you change it to `Auto-update documentation`?

**Review by bczsalba — COMMENTED** [review]
Very nice idea!

File: .github/workflows/generate_docs.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+          python-version: "3.x"
+
+      - name: Install pdoc
+        run: |
+          python -m pip install --upgrade pip
+          pip install pdoc
+
+      - name: Generate documentation
+        run: |
+          pdoc -o ./docs pytermgui
```
Comment (bczsalba) [inline .github/workflows/generate_docs.yml:30]: The command the project uses for documentation also includes these flags:
`--logo https://github.com/bczsalba/pytermgui/blob/master/assets/title.png?raw=true --docformat google`

**Comment by developomp** [comment]
done!

**Review by bczsalba — APPROVED** [review]
Thank you!

**Comment by developomp** [comment]
The workflow didn't run when the pr was merged because it only runs when there is a change in the `pytermgui` directory.

**Comment by bczsalba** [comment]
That's very useful, thank you!

**Pull request merged**